src/cnd/cnd.ts

```diff
@@ -329,6 +329,218 @@ export interface SwapProperties {
   [k: string]: any;
 }
 
+/**
+ * The payload returned when fetching one swap on the `/swaps/:id` endpoint
+ */
+export interface SwapResponse extends Entity {
+  properties?: FullSwapProperties;
+}
+
+/**
+ * Element of the array in the payload returned when fetching all swaps on the `/swaps/` endpoint
+ */
+export interface SwapElementResponse extends EmbeddedRepresentationSubEntity {
+  properties?: PartialSwapProperties;
+}
+
+/**
+ * The overall status of a swap
+ */
+export enum SwapStatus {
+  /**
+   * The swap was created but the communication phase hasn't yet been finalized
+   */
+  Created = "CREATED",
+  /**
+   * The communication was finalized and blockchain actions are needed or happening
+   */
+  InProgress = "IN_PROGRESS",
+  /**
+   * The swap is finished and the assets were swapped
+   */
+  Swapped = "SWAPPED",
+  /**
+   * The swap is finished and the assets were not swapped
+   */
+  NotSwapped = "NOT_SWAPPED",
+  /**
+   * An unexpected internal failure aborted the swap
+   */
+  InternalFailure = "INTERNAL_FAILURE"
+}
+
+/**
+ * The parameters for one side of the swap
+ */
+export interface Parameters {
+  // Do we want enums instead?
+  /**
+   * Protocol: "han", "herc20" or "halight"
+   */
+  protocol: string;
+  /**
+   * Ledger: "ethereum", "bitcoin" or "lightning"
+   */
+  ledger: string;
+  /**
+   * Network of the ledger: "mainnet", "testnet", "regtest" for Bitcoin, a chain id number for Ethereum
+   */
+  // This is tricky, an alternative would be to have more nesting where each ledger type (Bitcoin, Ethereum, Lightning) have a specific interface to declare network and asset
+  network: string | number;
+  /**
+   * Asset: "bitcoin", "ether" or { standard: "erc20", contract_address: "0x..." } for non-native tokens
+   */
+  // Not sure this is ideal, let's discuss and see what we can come up with
+  asset: string | { standard: "erc20"; contract_address: "0x..." };
```

Comment:
Instead of returning raw data, we could opt to format this for the user already. Likely, they will want to see something like "0.001 BTC" or "52530 PAY".

Reply:
Can anyone think of an application on top of COMIT that would need structured data here _after_ the swap has already started?
It doesn't have to be a string btw, we could go for something like:

`{
	"quantity": "52530",
	"symbol": "PAY"
}`

Reply:
We don't have to maintain a mapping for this (should likely cache it though). We can ask the ERC20 contract for it's symbol via `eth_call`.

Reply:
I think contract address for ERC20 tokens is required. The token metadata itself is optional since you can query it yourself, but having it included is helpful.

Reply:
At this stage I am re-using the existing `Asset` class as it contains the contract. I am not convinced cnd should be fetching such information from the blockchain. I think it makes more sense for cnd to provide the raw information upon which the protocol is executed.

Reply:
Well it is always a trade-off right. The reason I was suggesting this is because it simplifies the API, which is what we are trying to do here.

If we say, the contract address is required, then my question would be required for what?

I am fine if we can name usecases that will need the address for things that we cannot or don't want to provide as cnd.

> I am not convinced cnd should be fetching such information from the blockchain.

We can include a static list if we don't want to fetch the information. But the source is not what I would want to discuss. What I want to discuss is the reason we are including the information.

Any conditional logic that we can avoid for the client (`if asset.kind == 'erc20' { ... }`) is a win in terms of how easy the API is to use. Every polymorphic field in the response is going to cause some kind of conditional in the client ..

Reply:
@yosriady what's your input on this topic please? I consider it unresolved at this stage :)